### DISCUSSION TITLE
Startup steps

### DISCUSSION THREAD

**Discussion by markagama** [oqtane/oqtane.framework#1707] [Q&A]
Hi, I'm looking for a Blazor framework that integrates with SyncFusion Blazor components and provides authentication similar to Azure B2C - the website would be public facing and permit anyone to connect. I see the articles on how to get Syncfusion working with Oqtane.

I would appreciate guidance on the steps that I should take (what code to use) to move to the point where I can start to build and add modules that include Syncfusion.

**Comment by hishamco** [comment]
> Hi, I'm looking for a Blazor framework that integrates with SyncFusion Blazor components and provides authentication similar to Azure B2C - the website would be public facing and permit anyone to connect. I see the articles on how to get Syncfusion working with Oqtane.

@ADefWebserver already did some integrations and blog posts about SyncFusion integration

**Reply by markagama** [reply]
thx for the pointer, this was the article I mentioned in my post.

My question is what order should I do each activity, e.g., install oqtane, update to include syncfusion, what about Azure B2C or is Oqtane ok as it is now for authentication?

Also, articles get out of date quickly. I notice a couple of the key articles reference dotnet 3.1 and not dotnet 5. Are the articles going to be updated?

**Reply by hishamco** [reply]
> My question is what order should I do each activity, e.g., install oqtane, update to include syncfusion, what about Azure B2C or is Oqtane ok as it is now for authentication?

Did you checked all his blog posts?

> I notice a couple of the key articles reference dotnet 3.1 and not dotnet 5. Are the articles going to be updated?

This is not a big deal, you can update your projects then, to include the .NET 5 features & enhancements. I think you should target Oqtane version which is already target .NET version in your behalf

**Reply by markagama** [reply]
Hi Hishamco, I think I read somewhere back in the 1990s, that answering questions with a question on a forum about coding steps is not helpful. If you have advice on what steps I should take, then great, I would appreciate your advice on the steps. If your next step is to look at all of the blog posts, then don't bother. I've done this and as I mentioned, some of the posts/articles are out of date and there have been changes between dotnet 3.1 and 5 that can cause a lot of problems. I assume the same will happen with dotnet 6 and the new blazor - positive improvements we're all looking for.

**Reply by hishamco** [reply]
That's why I told you to target specific version of Oqtane, if a major change happen this means a breaking changes happen in the framework itself or it upgrades the .NET TFM

**Reply by markagama** [reply]
Hi Hishamco, I think you misunderstand my question. I would target the .net 5 code that is available now.
After I download this code, should I tackle integrating syncfusion next? I read that Azure B2C is not supported, this was in an early post. Is this still the case? Is Oqtane setup now to provide an alternative to Azure B2C? How do I get Microsoft accounts to work without Azure B2C?

thx

**Reply by markagama** [reply]
I should also mention that I've read about the changes to IdentityServer, hence why I was looking at Azure B2C, which I assume Microsoft will position to be an alternative to IdentityServer. I would appreciate knowing which way will Oqtane go over the next 12 months for user authentication.

**Reply by hishamco** [reply]
Please refer to this https://github.com/oqtane/oqtane.framework/discussions/827

**Reply by sbwalker** [reply]
@markagama Azure B2C is on the roadmap - however migrating to .NET 6 is the higher priority right now.

**Reply by markagama** [reply]
Thank you Shaun. I appreciate the history of DNN and the positive move to oqtane. I saw the roadmap and would like to ask when a mechanism to add/remove and update modules will be added?  It appears from my perspective that when Azure B2C or similar is added (pity about IdentityServer), the next most important aspect is to be able to build modules to add to framework. Hopefully automating this activity is a focus. I think the automation of the framework updates is really great, and something that most of the earlier generation of CMS could have done with. Updating drupal, e.g., can be a pain at times.
